### ISSUE TITLE
Problem with schemas that include a period "."

### ISSUE THREAD

**Issue by cawise** [nicholas-ross/SSMS-Schema-Folders#27]
We are using multi-part schemas ("sub-schemas", if you like) that include a dot to separate the "parent schema" from the "sub-schema" like `[Technology.Logging]`. In most cases, this extension works as expected. However, when the table name happens to be a prefix/substring of the "sub-schema", it groups weirdly -- as though the table name is removed from the schema name and leaves any remaining letters.

In the image below, you can see a few examples of this.
[Technology.***LandingPage***s].[LandingPage] yields "Technologys"
[Technology.***Log***ging].[Log] yields "Technologyging"

![image001](https://user-images.githubusercontent.com/39171616/181283835-8f7d3e87-bd6e-4d98-a468-f911af28b7c7.png)

**Comment by scottkuhl** [comment]
Should we expect a release with this fix in soon or are you waiting for more issues / features to add before doing one?  Just asking to determine if we should pull the code down and do our own internal release or wait for an official one.

Thanks for the quick fix!

**Comment by nicholas-ross** [comment]
I will be adding Folder Levels for the next release and need to update the localisation of the new features.

Since I'm not sure when it will be complete, I have done a [beta release](/nicholas-ross/SSMS-Schema-Folders/releases/tag/v1.4-beta) of the current build.

**Comment by cawise** [comment]
Beta works great! Thanks for the quick fix!

**Comment by cawise** [comment]
Here's a related issue, but I can open as a separate item if you like.

With these "multipart" schema names, when the Object Display Options > Rename Node option is set to True, it only strips off the first part of the schema name. 

![image](https://user-images.githubusercontent.com/39171616/182691428-e2cfbb32-986d-40e0-a429-e820ff5fcebf.png)

**Comment by nicholas-ross** [comment]
Thank you. I am aware of it.

It was a performance consideration I made at the time. To get the accurate object name a call to GetService for each node is required. I'm already doing this once but the rename was a different method and didn't want to repeat the same call. After merging in my folder level code I have decided it needs to be rewritten to parse all the folder levels and name with one call.
